### DISCUSSION TITLE
Call back with parameters

### DISCUSSION THREAD

**Discussion by coolH-alt** [fltk-rs/fltk-rs#1199] [Q&A]
i want my f callball with parameters(f32)

let on = self.value.clone();

self.f.handle({
            move |b, ev| match ev {
                Event::Drag => {
                    let t = app::event_y() as f32 / 100.0;
                    let mut num = t;
                    if t < 0.0 {
                        num = 0.0
                    }
                    if t > 1.0 {
                        num = 1.0
                    }
                    *on.borrow_mut() = num;
                    println!("ooooo {:?}", on);
                    b.do_callback();
                    b.redraw();
                    true
                }
                Event::Push => {
                    // println!("Push: {}", app::event_y());
                    true
                }
                Event::Released => {
                    // println!("Released: {}", app::event_y());
                    true
                }
                _ => false,
            }
        });

**Comment by MoAlyousef** [comment]
FLTK doesn't offer that. You would have to store the value in a smart pointer like you seem to be doing, then use it in the set_callback method.. So when you use do_callback(), it's invoked on the correct value.
